policy-controller/k8s/index/src/outbound_index.rs

```diff
@@ -0,0 +1,281 @@
+use crate::http_route::InboundRouteBinding;
+use ahash::AHashMap as HashMap;
+use anyhow::Result;
+use k8s_gateway_api::HttpBackendRef;
+use linkerd_policy_controller_core::{
+    http_route::{Backend, OutboundHttpRoute, OutboundHttpRouteRule, WeightedDst},
+    OutboundPolicy,
+};
+use linkerd_policy_controller_k8s_api::{
+    policy::{httproute::HttpRouteRule, HttpRoute},
+    ResourceExt, Service,
+};
+use parking_lot::RwLock;
+use std::{fmt, net::IpAddr, num::NonZeroU16, sync::Arc};
+use tokio::sync::watch;
+
+use super::http_route::convert;
+
+pub type SharedIndex = Arc<RwLock<Index>>;
+
+#[derive(Debug)]
+pub struct Index {
+    namespaces: NamespaceIndex,
+    services: HashMap<IpAddr, ServiceRef>,
```

Comment:
It looks like we're storing an index of services by namespace, and then a map of service name and nses by IP, so that we can look up an IP and find the service using that ns and name pair. When we were doing a prototype implementation a few weeks ago, I handled this a bit differently: I stored a map of IP addresses to `watch::Receiver`s, so that we skip the indirection of looking up the service by its name and ns and just return the watch that's already in that mapping.

Would an approach like make sense here? It could be a bit simpler conceptually, although, on the other hand, it would necessitate some reindexing logic that we don't currently have...

Not a blocker at all, just wanted to get your thoughts on that.